tests/components/plant/test_init.py

```diff
@@ -47,206 +46,193 @@
+    """Test bootstrapping the brightness history from the database.
+
+    This test can should only be executed if the loading of the history
+    is enabled via plant.ENABLE_LOAD_HISTORY.
+    """
+    init_recorder_component(hass)
+    plant_name = "wise_plant"
+    for value in [20, 30, 10]:
+
+        hass.states.async_set(
+            BRIGHTNESS_ENTITY, value, {ATTR_UNIT_OF_MEASUREMENT: "Lux"}
         )
-        self.hass.states.set(BRIGHTNESS_ENTITY, 100, {ATTR_UNIT_OF_MEASUREMENT: "lux"})
-        self.hass.block_till_done()
-        state = self.hass.states.get(f"plant.{plant_name}")
-        assert STATE_PROBLEM == state.state
-
-        self.hass.states.set(BRIGHTNESS_ENTITY, 600, {ATTR_UNIT_OF_MEASUREMENT: "lux"})
-        self.hass.block_till_done()
-        state = self.hass.states.get(f"plant.{plant_name}")
-        assert STATE_OK == state.state
-
-        self.hass.states.set(BRIGHTNESS_ENTITY, 100, {ATTR_UNIT_OF_MEASUREMENT: "lux"})
-        self.hass.block_till_done()
-        state = self.hass.states.get(f"plant.{plant_name}")
-        assert STATE_OK == state.state
-
-
-class TestDailyHistory(unittest.TestCase):
-    """Test the DailyHistory helper class."""
-
-    def test_no_data(self):
-        """Test with empty history."""
-        dh = plant.DailyHistory(3)
-        assert dh.max is None
-
-    def test_one_day(self):
-        """Test storing data for the same day."""
-        dh = plant.DailyHistory(3)
-        values = [-2, 10, 0, 5, 20]
-        for i in range(len(values)):
-            dh.add_measurement(values[i])
-            max_value = max(values[0 : i + 1])
-            assert 1 == len(dh._days)
-            assert dh.max == max_value
-
-    def test_multiple_days(self):
-        """Test storing data for different days."""
-        dh = plant.DailyHistory(3)
-        today = datetime.now()
-        today_minus_1 = today - timedelta(days=1)
-        today_minus_2 = today_minus_1 - timedelta(days=1)
-        today_minus_3 = today_minus_2 - timedelta(days=1)
-        days = [today_minus_3, today_minus_2, today_minus_1, today]
-        values = [10, 1, 7, 3]
-        max_values = [10, 10, 10, 7]
-
-        for i in range(len(days)):
-            dh.add_measurement(values[i], days[i])
-            assert max_values[i] == dh.max
+        await hass.async_block_till_done()
+    # wait for the recorder to really store the data
+    hass.data[recorder.DATA_INSTANCE].block_till_done()
+
+    assert await async_setup_component(
+        hass, plant.DOMAIN, {plant.DOMAIN: {plant_name: GOOD_CONFIG}}
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get(f"plant.{plant_name}")
+    assert STATE_UNKNOWN == state.state
+    max_brightness = state.attributes.get(plant.ATTR_MAX_BRIGHTNESS_HISTORY)
+    assert 30 == max_brightness
+
+
+async def test_brightness_history(hass):
+    """Test the min_brightness check."""
+    plant_name = "some_plant"
+    assert await async_setup_component(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5422434)